assembl/tests/model_tests/test_notifications.py

```diff
@@ -98,6 +98,64 @@ def test_notification_follow_synthesis(test_session, discussion,
     assert notification_count == initial_notification_count + 1, "The synthesis post should have matched and created a notification"
 
 
+def test_synthesis_notification_in_user_preferred_language_fr(test_session, discussion,
+                                                           participant1_user, participant1_user_language_preference_fr_cookie,
+                                                           test_app, root_post_1,
+                                                           synthesis_post_1):
+    test_session.flush()
+    subscription = NotificationSubscriptionFollowSyntheses(
+        discussion=discussion,
+        user=participant1_user,
+        creation_origin=NotificationCreationOrigin.USER_REQUESTED
+    )
+    test_session.add(subscription)
+
+    initial_notification_count = test_session.query(Notification).count()
+    dispatcher = ModelEventWatcherNotificationSubscriptionDispatcher()
+
+    notification_count = test_session.query(Notification).count()
+    
+    dispatcher.processPostCreated(synthesis_post_1.id)
+
+    last_notification = participant1_user.notifications[-1]
+
+    html_content = last_notification.render_to_email_html_part()
+
+    assert "subject FR" in html_content
+    assert "introduction FR" in html_content
+    assert "conclusion FR" in html_content
+    # TODO: also check that syntheses of several ideas do also show in the user's preferred language, and other gettext strings if available
```

Comment:
Can you `lxml` here to quickly parse the rendered tree and get the value of an idea in the preferred language set by test, and count the number of `div`s of ideas to verify that rendering is working.

```
from lxml import html
html_tree = html.fromstring(html_content)
```

From there, you can parse the tree according to the appropriate divs and make counts. More info:
http://lxml.de/lxmlhtml.html#html-element-methods

Reply:
Yes I wanted to do it in a second pass